lib/rules.js

```diff
@@ -197,9 +197,11 @@ module.exports = [
           style = styleForToken;
         }
         if ((styleForToken !== style) &&
-            !((style === "setext_with_atx") &&
+            !((style === "setext_with_atx" || style === "setext_with_atx_closed") &&
               (/h[12]/.test(token.tag) && (styleForToken === "setext")) ||
-               (/h[^12]/.test(token.tag) && (styleForToken === "atx")))) {
+              (/h[^12]/.test(token.tag) &&
+                  (styleForToken === "atx" || styleForToken === "atx_closed")))
```

Comment:
As I look at this more, I'm not even sure I followed my own guidelines with the original code. :) For line 203, I'd prefer it one space in from the previous line. And please don't spend any more time on indentation - I feel bad about the distraction!